src/Vacuum/Vacuum_init.jl

```diff
@@ -0,0 +1,198 @@
+        cnqd, snqd, sinlt, coslt, snlth, 
+        cslth, xplap, zplap = setuparrays!(
+            mth, dth, thgr, lmin, lmax,
+            qa1, n, settings.Shape.delfac,
+            delta, input.xpla, input.zpla
+        )
+
+
+    
+    
+
+    return VacuumGlobalsType(
+        mth=mth,
+        mth1=mth1,
+        mth2=mth2,
+        nfm=nfm,
+        mtot=mtot,
+        lmin=lmin,
+        lmax=lmax,
+        xinf=xinf,
+        zinf=zinf,
+        delta=delta,
+        xplap=xplap,
+        zplap=zplap,
+        qa1=qa1,
+        ga1=ga1,
+        fa1=fa1,
+        delx=delx,
+        delz=delz,
+        xwal=xwal,
+        zwal=zwal,
+        xwalp=xwalp,
+        zwalp=zwalp,
+        dth=dth,
+        wall=wall,
+        farwal=farwal,
+        kernelsign=kernelsign
+    )
+end
+
+function setuparrays!(
+    mth::Int,
+    dth::Float64,
+    thgr::Vector{Float64},
+    lmin::Vector{Int},
+    lmax::Vector{Int},
+    q::Float64,
+    n::Int,
+    delfac::Float64,
+    delta::Vector{Float64},
+    xpla::Vector{Float64},
+    zpla::Vector{Float64},
+)
+    # Sizes
+    mth1 = mth + 1
+    jmax1 = lmax[1] - lmin[1] + 1
+    nq = n * q
+
+    # Compute geometry bounds
+    xmin = minimum(xpla)
+    xmax = maximum(xpla)
+    # zmin = minimum(zpla)
+    # zmax = maximum(zpla)
+
+    plrad = 0.5 * (xmax - xmin)
+    delx = plrad * delfac
+    delz = plrad * delfac
+
+    #!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    # Wall shape (placeholder: straight line wall matching plasma)
+    # This is a placeholder for the wall shape, which should be replaced
+    # with the actual wall shape logic in wwall.
+    xwal = copy(xpla[1:mth1])
+    zwal = copy(zpla[1:mth1])
+    #!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+    # Derivatives (simple finite difference instead of spline)
+    # Will convert to spline once equil branch is merged
+    # xwalp = [ (xwal[mod1(i+1,mth)] - xwal[i]) / dth for i in 1:mth ]
+    # zwalp = [ (zwal[mod1(i+1,mth)] - zwal[i]) / dth for i in 1:mth ]
+
+    itp_x = Interpolations.interpolate((theta,), xwal, Interpolations.BSpline(Interpolations.Cubic(Interpolations.Periodic(OnGrid()))))
+    itp_z = Interpolations.interpolate((theta,), zwal, Interpolations.BSpline(Interpolations.Cubic(Interpolations.Periodic(OnGrid()))))
+
+    xwalp = [Interpolations.derivative(itp_x, 1, theta) for theta in thgr[:mth1]]
+    zwalp = [Interpolations.derivative(itp_z, 1, theta) for theta in thgr[:mth1]]
+
+    itp_x = Interpolations.interpolate((theta,), xpla, Interpolations.BSpline(Interpolations.Cubic(Interpolations.Periodic(OnGrid()))))
+    itp_z = Interpolations.interpolate((theta,), zpla, Interpolations.BSpline(Interpolations.Cubic(Interpolations.Periodic(OnGrid()))))
+
+    xplap = [Interpolations.derivative(itp_x, 1, theta) for theta in thgr[:mth1]]
+    zplap = [Interpolations.derivative(itp_z, 1, theta) for theta in thgr[:mth1]]
+
+    push!(xplap, xwalp[1])
+    push!(zplap, zwalp[1])
```

Comment:
This is an extra allocation as you have to increase the Vector size. If performance critical, you could preallocate `xplap` and `zplap` with vectors of `length(thgr[1:mth1])+1`, then populate the values.